### ISSUE TITLE
Lite: `shuffle=False` in DataLoader not working in DDP

### ISSUE THREAD

**Issue by awaelchli** [Lightning-AI/lightning#15930]
### Bug description

Lightning Lite reinstantiates the data loader when calling `lite.setup_dataloaders`, replacing the sampler with a DistributedSampler as expected. However, it also forces `shuffle=True`, regardless of the setting in the user's data loader. This is an artifact from Trainer which forces it by default during training. However, in Lite this does not make sense and the expected behavior would be to respect the choice of the user.

### How to reproduce the bug

```python
import torch
from torch.utils.data import DataLoader, TensorDataset

from lightning_lite import LightningLite


def run():
    lite = LightningLite(accelerator="cpu", strategy="ddp", devices=2)
    lite.launch()
    dataset = TensorDataset(torch.arange(10))
    dataloader = DataLoader(dataset, shuffle=False)
    dataloader = lite.setup_dataloaders(dataloader)

    if lite.global_rank == 0:
        print("rank 0")
        for data in dataloader:
            print(data)

    lite.barrier()

    if lite.global_rank == 1:
        print("rank 1")
        for data in dataloader:
            print(data)


if __name__ == "__main__":
    run()
```

### Error messages and logs

```
rank 0
[tensor([4])]
[tensor([7])]
[tensor([3])]
[tensor([0])]
[tensor([6])]
rank 1
[tensor([1])]
[tensor([5])]
[tensor([9])]
[tensor([8])]
[tensor([2])]
```
Expected: 0, 2, 4, 6, 8 and 1, 3, 5, 7, 9

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow): Lite
#- PyTorch Lightning Version (e.g., 1.5.0): 1.9.0dev
#- Lightning App Version (e.g., 0.5.2): x
#- PyTorch Version (e.g., 1.10): 1.13.0
#- Python version (e.g., 3.9): 3.10
#- OS (e.g., Linux): MacOS
#- CUDA/cuDNN version: x
#- GPU models and configuration: x
#- How you installed Lightning(`conda`, `pip`, source): x
#- Running environment of LightningApp (e.g. local, cloud): x
```

</details>

cc @carmocca @justusschock @awaelchli

**Comment by FarzanT** [comment]
Hi,
I wanted to add that a similar behavior persists when using DeepSpeed. Even when using a SequentialSampler, the passed index to the DataSet is still randomized.

**Comment by awaelchli** [comment]
@FarzanT The linked pull request should fix this in general. It was released as part of 1.8.4. If the issue persists, would you mind opening a new bug report with the details of your case? Thanks
